hal/common/time_keeper.hpp

```diff
@@ -49,7 +49,6 @@ extern "C" {
 

```

Comment:
doc and ifndef not matching,

Besides, why is there 
`#ifdef __cplusplus`
 `extern "C" {`
 `#endif`